### PR TITLE
.net - compileToBinary fix

### PR DESCRIPTION
**Pull request by partouf**
Removes shared variable that would conflict with simultaneous compilations

Also ups pids max because of an issue I was hitting locally

### CODE REVIEW

**Comment by kasperk81** [comment]
we shouldn't force `Main` method for compile mode.  generic method (M2 in example below) somehow require `Main`

image

adding empty `Main` fix it but it is definitely a bug with generic method
image

basically `<OutputType>Exe</OutputType>` in project file for execution mode _only_.

**Comment by hez2010** [comment]
> we shouldn't force `Main` method for compile mode.  generic method (M2 in example below) somehow require `Main`
> 
> image
> 
> adding empty `Main` fix it but it is definitely a bug with generic method
> image
> 
> basically `<OutputType>Exe</OutputType>` in project file for execution mode _only_.
> 
> 

For execution, you must provide an entry point. If you don't want to provide an entry point, you can choose not to compile to binary.

**Comment by kasperk81** [comment]
> For execution, you must provide an entry point. If you don't want to provide an entry point, you can choose not to compile to binary.

that's my point, i'm not using execution

**Comment by kasperk81** [comment]
https://godbolt.org/z/9deeWGfGc there is no execution. output window is for compile, not execution and yet it is throwing `CS5001`

**Comment by hez2010** [comment]
> https://godbolt.org/z/9deeWGfGc there is no execution. output window is for compile, not execution and yet it is throwing `CS5001`

That's weird. We should have set the `OutputType` to `Library` when compileToBinary is not checked.

**Comment by partouf** [comment]
> > https://godbolt.org/z/9deeWGfGc there is no execution. output window is for compile, not execution and yet it is throwing `CS5001`
> 
> That's weird. We should have set the `OutputType` to `Library` when compileToBinary is not checked.

that's because in live without this fix, the state is shared and the compileToBinary setting is never reset to false anymore

**Comment by hez2010** [comment]
Thanks for the fix. Looks good to me.

File: etc/nsjail/execute.cfg [hunk 1]
```diff
@@ -28,7 +28,7 @@ cgroup_net_cls_parent: "ce-compile"
 cgroup_cpu_parent: "ce-compile"
 
 cgroup_mem_max: 1342177280 # 1.25 GiB
-cgroup_pids_max: 64  # go uses a bunch (probably one per CPU, on my desktop)
+cgroup_pids_max: 72  # go uses a bunch (probably one per CPU, on my desktop)
```
Comment (mattgodbolt) [inline etc/nsjail/execute.cfg:31]: Can you updae the comment...or remove it? super low priority nit

**Review by mattgodbolt — APPROVED** [review]
Thanks mate!

**Review by mattgodbolt — COMMENTED** [review]
Can you fix the lint issues too please

File: lib/compilers/dotnet.ts [hunk 2]
```diff
@@ -105,29 +142,15 @@ class DotNetCompiler extends BaseCompiler {
...
-                <OutputType>${this.compileToBinary ? 'Exe' : 'Library'}</OutputType>
-            </PropertyGroup>
-            <ItemGroup>
-                <Compile Include="${sourceFile}" />
-            </ItemGroup>
-         </Project>
-        `;
+
+        const sdkBaseDir = path.join(path.dirname(compiler), 'sdk');
+        const sdkVersions = await fs.readdir(sdkBaseDir);
```
Comment (kasperk81) [inline lib/compilers/dotnet.ts:153]: unused

Comment (partouf) [reply lib/compilers/dotnet.ts:153]: ah. right, merge conflict things

Comment (kasperk81) [reply lib/compilers/dotnet.ts:153]: btw, how long does it take for changes to go live after the merge?

Comment (partouf) [reply lib/compilers/dotnet.ts:153]: It's a manual thing we have to do, push some buttons, test on testing environment, make fixes, deploy to live, etc. So somewhere between an hour and 2 weeks depending on the admins schedules.

**Comment by hez2010** [comment]
I guess we can just skip running crossgen2 if we are in execution-only mode?

**Comment by partouf** [comment]
> I guess we can just skip running crossgen2 if we are in execution-only mode?

Probably. You'll need to copy the `runCompiler` function, name it something else and call it from `buildExecutable`

**Comment by partouf** [comment]
this is now live

**Comment by kasperk81** [comment]
with yesterday's container, it seems. latest one is https://github.com/compiler-explorer/compiler-workflows/actions/runs/3569737898

**Comment by partouf** [comment]
installation happens at a later point during the night, in about 5 hours

**Comment by kasperk81** [comment]
> in about 5 hours

it wasn't updated. still using the one from day before yesterday

**Comment by partouf** [comment]
> > in about 5 hours
> 
> it wasn't updated. still using the one from day before yesterday

Latest is now installed. There were a couple things wrong with the build and install script

**Pull request merged**